File: Coffee/Models/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Coffee.Models
{
    public class Product
    {
        public int Id { get; set; }
        public int BrandId { get; set; }
        public string Description { get; set; }
        public double Price { get; set; }
        public string Image { get; set; }
        
        public List<Category> Categories { get; set; }
    }
}

File: Coffee/ClientApp/src/app/models/product.ts
import { Category } from "./Category";

export class Product {
  public id: number;
  public brandId: number;
  public description: string;
  public price: number;
  public image: string;
  public brand: string;
  public categories: Category[];
  constructor() { };
  public init(id: number, brandId: number, description: string, price: number, image: string) {
    this.id = id;
    this.brandId = brandId;
    this.description = description;
    this.price = price;
    this.image = image;
  }
}
